setup

```diff
@@ -742,10 +742,15 @@ def configure_default_compilers(args):
             args.fc  and os.path.basename(args.fc).lower().startswith('openmpi'):
             args.mpi = 'on'
 
-    if not args.cc:
-        args.cc = 'gcc'
-    if not args.cxx:
-        args.cxx = 'g++'
+    if not args.cc and os.environ['CC']:
```

Comment:
These dictionary lookups will throw `KeyError` if there is no `--cxx/cc/fc` argument passed and the environmental variable is not set. 

It should probably catch that and either default to gcc/g++ (like before) or give some kind of message like:

> No C/C++/FORTRAN  compiler detected!   
 Set the environmental variable $CC/CXX/FC or use the --cc/cxx/fc flag at the command line